### PR TITLE
timer added

### LINKED ISSUE #177: Show elapsed after sending a request
**Tell us about the task you want to perform and are unable to do so because the feature is not available**

Currently, when one sends a request, there is no timer displayed which shows how much time has elapsed. Only after the response is received is when the elapsed time is displayed. Through this feature, one can add a timer which continuously updates the elapsed time as the response is being awaited.

https://github.com/foss42/apidash/assets/1382619/67941a34-6314-4323-8d0f-8128e50cdd94

### PR DESCRIPTION
**Pull request by BrawlerXull**
Fixes #177

### CODE REVIEW

**Comment by ashitaprasad** [comment]
@BrawlerXull The timer component UI is currently not going well with the theme of the entire app. Please improve the UI and update the PR.

**Comment by ashitaprasad** [comment]
@BrawlerXull Check out this utility we already have - [link](https://github.com/foss42/apidash/blob/d328a925c71d02cbd43a7d4b5fed6b1fdbe1b42d/lib/utils/convert_utils.dart#L7) to make the time more human readable.

**Comment by animator** [comment]
Sure @BrawlerXull I'll take a look at it.

File: test/widgets/response_widgets_test.dart
Comment (ashitaprasad) [inline test/widgets/response_widgets_test.dart:1]: testWidgets('SendingWidget displays timer and Lottie animation') has been nested inside another testWidgets('Testing Body Success tap segment for formattedBody null') which is causing an error.

**Comment by ashitaprasad** [comment]
Hi @BrawlerXull, Testing is an important concept and I recommend you to kindly take a couple of days, go through the documentation to understand how to write tests in Flutter.

**Comment by ashitaprasad** [comment]
@BrawlerXull The implementation has a bug. If the user moves to another request, and then returns then the timer starts from 0 again and does not show the actual time that has elapsed so far. Please correct this issue.

**Comment by animator** [comment]
@BrawlerXull any updates?

**Comment by ashitaprasad** [comment]
Please open a new PR with the correct approach.

**Pull request closed without merging**